binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,71 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        # adds the input value to the binary search tree, adhering to the rules of the ordering of elements in a binary search tree.
+        # Dont forget to wrap the value in a node
+        # 1. compare the element against the current node's value
+        # 2. based on the result of the comparison, go left or right
+        # 3. if we find an empty spot, park the value there
+        # 4. otherwise, go back to step 1
 
-  def contains(self, target):
-    pass
+        # What is the base case?
+        # base case: we've found an empty spot where we can add the value
+        if value < self.value:
+            # if value is less, we go left
+            # if there isnot left child, we can park the node here
+            if not self.left:
+                self.left = BinarySearchTree(value)
```

Comment:
Excellent, grow the tree if necessary. What would you do if this tree starts to become unbalanced?